presto-parser/src/test/java/com/facebook/presto/sql/parser/TestSqlParser.java

```diff
@@ -1645,19 +1651,63 @@ public void testExists()
     }
 
     @Test
-    public void testShowColumnStats()
+    public void testShowStats()
     {
         final String[] tableNames = {"t", "s.t", "c.s.t"};
 
         for (String fullName : tableNames) {
             QualifiedName qualifiedName = QualifiedName.of(Arrays.asList(fullName.split("\\.")));
             assertStatement(format("SHOW STATS %s", qualifiedName), new ShowStats(qualifiedName));
-            assertStatement(format("SHOW STATS FROM %s", qualifiedName), new ShowStats(qualifiedName));
-            assertStatement(format("SHOW STATS IN %s", qualifiedName), new ShowStats(qualifiedName));
+            assertStatement(format("SHOW STATS FOR %s", qualifiedName), new ShowStats(qualifiedName));
+            assertStatement(format("SHOW STATS ON %s", qualifiedName), new ShowStats(qualifiedName));
         }
     }
 
     @Test
+    public void testShowStatsParametrized()
+    {
+        final String[] tableNames = {"t", "s.t", "c.s.t"};
+
+        for (String fullName : tableNames) {
+            QualifiedName qualifiedName = QualifiedName.of(Arrays.asList(fullName.split("\\.")));
+            assertStatement(format("SHOW STATS FOR (SELECT * FROM %s)", qualifiedName),
+                    createShowStats(qualifiedName, ImmutableList.of(new AllColumns()), Optional.empty()));
+            assertStatement(format("SHOW STATS FOR (SELECT * FROM %s WHERE field > 0)", qualifiedName),
+                    createShowStats(qualifiedName,
+                            ImmutableList.of(new AllColumns()),
+                            Optional.of(
+                                    new ComparisonExpression(GREATER_THAN,
+                                            new QualifiedNameReference(QualifiedName.of("field")),
+                                            new LongLiteral("0")))));
+            assertStatement(format("SHOW STATS FOR (SELECT * FROM %s WHERE field > 0 or field < 0)", qualifiedName),
+                    createShowStats(qualifiedName,
+                            ImmutableList.of(new AllColumns()),
+                            Optional.of(
+                                    new LogicalBinaryExpression(OR,
+                                        new ComparisonExpression(GREATER_THAN,
+                                                new QualifiedNameReference(QualifiedName.of("field")),
+                                                new LongLiteral("0")),
+                                            new ComparisonExpression(LESS_THAN,
+                                                    new QualifiedNameReference(QualifiedName.of("field")),
+                                                    new LongLiteral("0"))
+                                            )
+                                    )));
+        }
+    }
+
+    private ShowStats createShowStats(QualifiedName name, List<SelectItem> selects, Optional<Expression> where)
+    {
+        return new ShowStats(new QuerySpecification(
```

Comment:
This would require to transit ShowStats from usage of QuerySpecification to Query, though it might be a good idea.